distributed/utils.py

```diff
@@ -941,9 +941,8 @@ def truncate_exception(e, n=10000):
 
 def validate_key(k):
     """Validate a key as received on a stream."""
-    typ = type(k)
-    if typ is not str and typ is not bytes:
-        raise TypeError(f"Unexpected key type {typ} (value: {k!r})")
+    if not isinstance(k, (bytes, int, float, str, tuple)):
```

Comment:
I think this should recursively validate the elements of `k` if `k` is a `tuple`.